### PR TITLE
All user tests done.

### PR DESCRIPTION
**Pull request by razashtar93**
Added:
1) test for create user
2) test for delete user
3) test for get user

All user tests pass

there is also:
pipenv install --dev pytest-django
pytest.ini file

#55 #56 

@Yarboa 
if all looks good can this PR overwrite  #55 #56?

### CODE REVIEW

File: .idea/workspace.xml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Yarboa) [inline .idea/workspace.xml:1]: Please git rm --cache .idea 
this directory before next commit

You can always do a git status before git push, 
and use git rm --cache to the files or dirs you want to exclude

File: AcadeMeData/tests.py [hunk 2]
```diff
@@ -1,35 +1,40 @@
...
-        user_data = {'username': "username", 'password': "password", 'email': "user@example.com", 'type': "S",
-                     'university': "RU",
-                     'degree': "CS"}
-        user = User.create_user(*user_data)
-        return user, user_data
+        data_for_user = {'username': "username", 'password': "password", 'email': "user@example.com", 'type': "S",
+                         'university': "RU",
+                         'degree': "CS"}
+        user = User.create_user(*data_for_user)
+        return user
```
Comment (Yarboa) [inline AcadeMeData/tests.py:11]: return date_for_user,
the user created and in the database

File: AcadeMeData/tests.py [hunk 3]
```diff
@@ -1,35 +1,40 @@
...
-        assert isinstance(user, User)
-        assert user.username == user_data.get('username')
-        assert user.password == user_data.get('password')
-        assert user.email == user_data.get('email')
-        assert user.type == user_data.get('type')
-        assert user.university == user_data.get('university')
-        assert user.degree == user_data.get('degree')
-        assert User.get_user(user_data.get('username')).degree == user_data.get('degree')
+        users_list = User.objects.all()
+        user_data = self.user_example()
```
Comment (Yarboa) [inline AcadeMeData/tests.py:16]: this should be before:

1st creating user
2nd verify it is created

File: AcadeMeData/tests.py [hunk 4]
```diff
@@ -1,35 +1,40 @@
...
-class TestUserModel(pytest):
+class TestUserModel:
     def user_example(self):
-        user_data = {'username': "username", 'password': "password", 'email': "user@example.com", 'type': "S",
-                     'university': "RU",
-                     'degree': "CS"}
-        user = User.create_user(*user_data)
-        return user, user_data
+        data_for_user = {'username': "username", 'password': "password", 'email': "user@example.com", 'type': "S",
+                         'university': "RU",
```
Comment (Yarboa) [inline AcadeMeData/tests.py:8]: please delete spaces, it will give you flake8 error

File: AcadeMeData/tests.py [hunk 5]
```diff
@@ -1,35 +1,40 @@
...
 
-    @pytest
+        assert users_list[len(users_list) - 1] == user_data  # is this enough? only check if the objects are equal
+        # if the line above enough then we can delete the first 3 assert line (username, email,password)
+
+        # assert users_list[len(users_list) - 1].user.type == user_data.type  # this now working
+        # assert users_list[len(users_list) - 1].user.university == user_data.university # this now working
+        # assert users_list[len(users_list) - 1].user.degree == user_data.degree # this now working
+
+    @pytest.mark.django_db
```
Comment (Yarboa) [inline AcadeMeData/tests.py:28]: this should be for the whole class, move to line 4

File: AcadeMeData/tests.py [hunk 6]
```diff
@@ -1,35 +1,40 @@
...
     def test_del_user(self):
-        user, user_data = self.user_example()
-        assert isinstance(user, User)
-        assert User.del_user(user)
-        assert not (User.get_user(user_data.get('username')))
+        users_list = User.objects.all()
+        user_data = self.user_example()
+        assert User.del_user(user_data)
+        assert not users_list[len(users_list) - 1] == user_data
```
Comment (Yarboa) [inline AcadeMeData/tests.py:33]: see the check for the create user, not like that

File: AcadeMeData/tests.py [hunk 7]
```diff
@@ -1,35 +1,40 @@
...
 
-    @pytest
+    @pytest.mark.django_db
```
Comment (Yarboa) [inline AcadeMeData/tests.py:35]: this one is a duplication of the create user,
see my comments, search already tested there

File: AcadeMeData/tests.py [hunk 8]
```diff
@@ -1,35 +1,40 @@
...
-        assert user.username == user_data.get('username')
-        assert user.password == user_data.get('password')
-        assert user.email == user_data.get('email')
-        assert user.type == user_data.get('type')
-        assert user.university == user_data.get('university')
-        assert user.degree == user_data.get('degree')
-        assert User.get_user(user_data.get('username')).degree == user_data.get('degree')
+        users_list = User.objects.all()
+        user_data = self.user_example()
+        assert users_list[len(users_list) - 1].user.username == user_data.user.username
```
Comment (Yarboa) [inline AcadeMeData/tests.py:17]: It should be like that 
user_data returned from the user_example method, 
no searching in db after you have the search commend

user=User.get_user(user_data[usename])
It returns the whole object

now you have methods of User of django user

assert user.get_username() == user_data[usename]
assert user.get_email() == user_data[email]

**Note:**
you can tests the code by running the python from the /vagrant dir
pipenv run python manage.py shell
from AcadeMeData.models import User

and test every thing, tab will reveal the methods access in the classes

File: Pipfile [hunk 9]
```diff
@@ -5,6 +5,8 @@ verify_ssl = true
 
 [dev-packages]
 flake8 = "*"
+pytest = "*"
```
Comment (Yarboa) [inline Pipfile:8]: this is not needed
once you have django pytest, 
you can run it locally to test it

File: AcadeMeData/tests.py (hunk 4 above)
Comment (razashtar93) [reply AcadeMeData/tests.py:8]: when the line is too big this is the only way it pass flake8

File: AcadeMeData/tests.py (hunk 5 above)
Comment (razashtar93) [reply AcadeMeData/tests.py:28]: OK, I've changed it now

File: AcadeMeData/tests.py (hunk 8 above)
Comment (Yarboa) [reply AcadeMeData/tests.py:17]: This should work like this 
       user = User.create_user(*user_data)
        user = User.get_user("username")
        assert user.get_username() ==  user_data['username']


tested

Comment (razashtar93) [reply AcadeMeData/tests.py:17]: image
image

this what i got. what should user_example return?
@Yarboa

**Review by Yarboa — APPROVED** [review]
LGTM Nice work

**Pull request merged**